src/main/java/com/ibm/cloud/sdk/core/http/HttpClientSingleton.java

```diff
@@ -317,6 +343,15 @@ public OkHttpClient configureClient(OkHttpClient client, HttpConfigOptions optio
                         , options.getMaxRetries()))
                 .build();
       }
+      if (options.shouldEnableGzipCompression()) {
```

Comment:
I know this is sort of pathological, but what would happen if the user called "enableGzipCompression(true)" twice?  :)

One way to harden this piece of code that either adds or removes the interceptor would be to unconditionally remove the Gzip interceptor, then if "shouldEnableGzipCompression()" is true, then add it.   That way, we're guaranteed to have the Gzip interceptor registered IFF shouldEnableGzipCompression() is true and it will be present in the list at most once.